scripts/deploy.sh

```diff
@@ -7,17 +7,88 @@ pull_latest() {
+  fi
+  mkdir -p $build_folder
+
+  # We want latest commit hash, so we do a shallow copy of the repositories (and not simply a wget)
+  git clone --depth=1 https://github.com/cubing/wca-regulations.git $build_folder/wca-regulations
+  git clone --depth=1 https://github.com/cubing/wca-regulations-translations.git $build_folder/wca-regulations-translations
+  git_reg_hash=`git -C $build_folder/wca-regulations rev-parse --short HEAD`
+  git_translations_hash=`git -C $build_folder/wca-regulations-translations rev-parse --short HEAD`
+
+  rebuild_regulations=1
+  rebuild_translations=1
+  # Check if the cloned regulations match the current version
+  if [ -r $regs_version ] && [ "`cat $regs_version`" == "$git_reg_hash" ]; then
+    rebuild_regulations=0
+  fi
+  # Check if the cloned translations match the current version
+  if [ -r $translations_version ] && [ "`cat $translations_version`" == "$git_translations_hash" ]; then
+    rebuild_translations=0
+  fi
+  if [ $rebuild_regulations -eq 0 ] && [ $rebuild_translations -eq 0 ]; then
+    echo "WCA Regulations and translations are up to date."
+    return
+  fi
+
+  # Else we have to rebuild something
+
+  # This save tracked files that may have unstashed changes too
+  cp -r $regs_folder $build_folder
+
+  # Checkout data (scramble programs, history)
+  git checkout origin/regulations-data $build_folder
+  git reset HEAD $build_folder
+
+  languages=`wrc-languages`
+  if [ $rebuild_translations -eq 1 ]; then
+    # Clean up translations directories
+    find $build_folder/regulations/translations ! -name 'translations' -type d -exec rm -rf {} +
+    # Rebuild all translations
+    for kind in html pdf; do
+      for l in $languages; do
+        inputdir=$build_folder/wca-regulations-translations/${l}
+        outputdir=$build_folder/regulations/translations/${l}
+        mkdir -p $outputdir
+        echo "Doing "${kind}" for language "${l}
+        wrc --target=$kind -l $l -o $outputdir -g $git_translations_hash $inputdir
+      done
+    done
+    # Update version built
+    echo $git_translations_hash > $build_folder/regulations/translations/version
+  else
+    echo "Translations are up to date."
+  fi
+
+  outputdir=$build_folder/regulations
+  if [ $rebuild_regulations -eq 1 ]; then
+    # Clean up regulations directory files
+    find $build_folder/regulations -maxdepth 1 -type f -exec rm -f {} +
+    # Rebuild Regulations
+    wrc --target=json -o $outputdir -g $git_reg_hash $build_folder/wca-regulations
+    wrc --target=html -o $outputdir -g $git_reg_hash $build_folder/wca-regulations
+    wrc --target=pdf -o $outputdir -g $git_reg_hash $build_folder/wca-regulations
+    # Update version built
+    echo $git_reg_hash > $build_folder/regulations/version
+  else
+    echo "Regulations are up to date"
+  fi
+
   if [ -a $regs_folder ]; then
-    rm -rf $regs_folder-todelete
-    mv $regs_folder $regs_folder-todelete
+    # Remove previous build
+    rm -rf $regs_folder
```

Comment:
Interesting, I messed that point, and when restoring the convoluted version I got that:

> /vagrant @ production> rm -rf WcaOnRails/app/views/regulations-todelete
> rm: cannot remove ‘WcaOnRails/app/views/regulations-todelete/scrambles’: Directory not empty
> rm: cannot remove ‘WcaOnRails/app/views/regulations-todelete/history’: Directory not empty
> rm: cannot remove ‘WcaOnRails/app/views/regulations-todelete/translations’: Directory not empty
> /vagrant @ production> ls -al WcaOnRails/app/views/regulations-todelete/
> total 20
> drwxr-xr-x 1 vagrant vagrant 4096 Sep  6 12:11 .
> drwxr-xr-x 1 vagrant vagrant 4096 Sep  6 12:11 ..
> drwxr-xr-x 1 vagrant vagrant 4096 Sep  6 12:11 history
> drwxr-xr-x 1 vagrant vagrant 4096 Sep  6 12:11 scrambles
> drwxr-xr-x 1 vagrant vagrant 4096 Sep  6 12:11 translations
> /vagrant @ production> ls -al WcaOnRails/app/views/regulations-todelete/scrambles/
> total 8
> drwxr-xr-x 1 vagrant vagrant 4096 Sep  6 12:11 .
> drwxr-xr-x 1 vagrant vagrant 4096 Sep  6 12:11 ..

The funny part is when I list the same directory from my computer:

> [14:27] daisy:~/Documents/cube/wca/VM/worldcubeassociation.org (merged|✚7)$ lal WcaOnRails/app/views/regulations-todelete/scrambles 
> total 20
> drwxr-xr-x 3 fifi fifi 4096 sept.  6 14:04 .
> drwxr-xr-x 5 fifi fifi 4096 sept.  6 14:11 ..
> -rw-r--r-- 1 fifi fifi 4107 sept.  6 14:04 index.html.erb
> drwxr-xr-x 3 fifi fifi 4096 sept.  6 14:04 tnoodle


So `rm -rf` fails at removing an empty dir saying it's not empty... Sounds like yet another issue with the shared directories between vagrant and my computer.
I have no clue what's going on for this particular directory, I can remove other parts from the shared directories correctly...

Therefore I moved the "todelete" folder to some local directory in `/tmp`.